ansible/verify_config.py

```diff
@@ -0,0 +1,768 @@
+                                       reason=f"{param['name']} devices does not exist in its devices file")
+
+    def _links_file_should_have_equivalent_devices_file(self, param):
+        inv_name_from_links_files = Utility.get_inv_name_from_file(param["file"])
+        inv_name_from_devices_files = Utility.get_inv_name_from_file(Config.FANOUT_DEVICES_FILE)
+
+        differences = inv_name_from_links_files ^ inv_name_from_devices_files
+
+        if differences:
+            self.assertion.add_error_details(
+                f"These are the group names that do not have devices file: [{Formatting.red(', '.join(differences))}]. "
+                "Consider creating "
+                f"[{', '.join(Formatting.bold(f'files/sonic_{group_name}_devices.csv') for group_name in differences)}]"
+            )
+            return False
+
+        return True
+
+    def _devices_in_links_file_should_be_in_devices_file(self, param):
+        is_valid = True
+        inv_name_from_links_files = Utility.get_inv_name_from_file(param["file"])
+
+        for group_name in inv_name_from_links_files:
+            device_file = Config.FANOUT_DEVICES_FILE.replace("*", group_name)
+            link_file = param['file'].replace("*", group_name)
+            devices_from_links_file = Utility.get_devices_from_links_file(link_file)
+            devices_from_devices_file = Utility.get_devices_from_devices_file(device_file)
+
+            differences = devices_from_links_file - devices_from_devices_file
+
+            if differences:
+                self.assertion.add_error_details(
+                    "These are the devices that are in "
+                    f"{Formatting.yellow(param['file'].replace('*', group_name))} "
+                    f"but not in devices file: [{Formatting.red(', '.join(differences))}]. "
+                    f"Consider adding in {Formatting.bold(device_file)}")
+                is_valid = False
+
+        return is_valid
+
+
+class HostNetworkValidation(Validator):
+    def __init__(self):
+        super().__init__()
+
+    def validate(self):
+        self.assertion.assert_true(
+            lambda: self._check_if_bridge_is_up(),
+            reason=f"Interface 'br1' is not up. Consider running "
+                   f"'{Formatting.yellow('./setup-management-network.sh')}'",
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_proxy_is_using(),
+            reason="Proxy setting is not correct ",
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_can_connect_to_docker_registry(),
+            reason="Cannot establish a connection to docker registry. Check your proxy setting, docker proxy setting"
+        )
+
+        self.assertion.assert_true(
+            lambda: self._check_if_can_connect_to_apt_repository(),
+            reason="Cannot establish a connection to apt repository. Check your proxy setting."
```

Comment:
Same comments with the docker-registry notice.
Proxy setting is not the only one reason of connectivity issues.
We need to ask the user to check the network, and setup http proxy if needed rather than ask them to check proxy settings directly

Reply:
Good point. I've adjusted to `"Cannot establish a connection to apt repository. Check your network connection."`